packages/driver/cypress/integration/e2e/multi-domain/multi_domain_uncaught_errors_spec.ts

```diff
@@ -0,0 +1,177 @@
+  beforeEach(() => {
+    cy.visit('/fixtures/multi-domain.html')
+    cy.get('a[data-cy="errors-link"]').click()
+  })
+
+  describe('uncaught errors', () => {
+    describe('sync errors', () => {
+      it('fails the current test/command if sync errors are thrown from the switchToDomain context', () => {
+        const uncaughtExceptionSpy = cy.spy()
+        const r = cy.state('runnable')
+
+        cy.on('uncaught:exception', uncaughtExceptionSpy)
+        cy.on('fail', (err, runnable) => {
+          // TODO: we likely need to change the messaging around this error to make it clear to multi-domain users that
+          // this behavior is configurable with 'uncaught:exception', but it MUST be declared inside the switchToDomain callback
+          // and that 'uncaught:exception' will NOT be triggered outside that callback (inside the primary domain)
+          expect(err.name).to.eq('Error')
+          expect(err.message).to.include('sync error')
+          expect(err.message).to.include('The following error originated from your application code, not from Cypress.')
+          expect(err.message).to.not.include('https://on.cypress.io/uncaught-exception-from-application')
+          // @ts-ignore
+          expect(err.docsUrl).to.deep.eq(['https://on.cypress.io/uncaught-exception-from-application'])
+
+          // lastly, make sure the `uncaught:exception' handler is NOT called in the primary
+          expect(uncaughtExceptionSpy.notCalled).to.be.true
+          expect(runnable === r).to.be.true
+        })
+
+        cy.switchToDomain('foobar.com', () => {
+          cy.get('.trigger-sync-error').click()
+        })
+      })
+
+      it('returns false from cy.on(uncaught:exception), resulting in cy:fail to not be called in the primary and passes the test', () => {
+        const uncaughtExceptionSpy = cy.spy()
+        const failureSpy = cy.spy()
+
+        cy.on('uncaught:exception', uncaughtExceptionSpy)
+
+        cy.on('fail', failureSpy)
+
+        cy.switchToDomain('foobar.com', () => {
+          cy.on('uncaught:exception', () => false)
+          cy.get('.trigger-sync-error').click()
+        }).then(() => {
+          expect(uncaughtExceptionSpy.notCalled).to.be.true
+          expect(failureSpy.notCalled).to.be.true
+        })
+      })
+
+      it('returns true from cy.on(uncaught:exception), resulting in cy:fail to be called in the primary', (done) => {
+        cy.on('fail', () => {
+          done()
+        })
+
+        cy.switchToDomain('foobar.com', () => {
+          cy.once('uncaught:exception', () => true)
+          cy.get('.trigger-sync-error').click()
+        })
+      })
+
+      // if we mutate the error, the app's listeners for 'error' or
+      // 'unhandledrejection' will have our wrapped error instead of the original
+      it('original error is not mutated for "error" in the switchToDomain', () => {
```

Comment:
Not sure if this test is extremely valuable here. I pulled it from the [uncaught-error](https://github.com/cypress-io/cypress/blob/issue-19709-multi-domain-handle-uncaught-exceptions/packages/driver/cypress/integration/e2e/uncaught_errors_spec.js#L119) spec and adapted it to multi-domain